cli/src/utilities/generateNWUI.ts

```diff
@@ -0,0 +1,58 @@
+import { GluegunToolbox } from 'gluegun';
+import { CliResults } from '../types';
+import { nativeWindUIOptions } from '../constants';
+import { getPackageManagerRunnerX } from './getPackageManager';
+import { ONLY_ERRORS, runSystemCommand } from './systemCommand';
+import { spinner } from '@clack/prompts';
+
+export async function generateNWUI(cliResults: CliResults, toolbox: GluegunToolbox) {
+  const isNativeWindUISelected = cliResults.packages.some((p) => p.name === 'nativewindui');
+
+  if (!isNativeWindUISelected) {
+    return;
+  }
+
+  const s = spinner();
+
+  const runnerType = getPackageManagerRunnerX(toolbox, cliResults);
+
+  const nativeWindUIComponents =
+    cliResults.packages.find((p) => p.name === 'nativewindui').options.selectedComponents ?? [];
+
+  // we do this to account for older stored config e.g that has selectable text in it
+  const onlySupportedComponents = nativeWindUIComponents.filter((component) => nativeWindUIOptions.includes(component));
+
+  const finalComponents = Array.from(new Set([...onlySupportedComponents, 'text']));
+
+  s.start(`Adding nativewindui components...`);
+
+  const flags = cliResults.flags.noInstall
+    ? `--yes --no-install -d ${cliResults.projectName}`
+    : `--yes -d ${cliResults.projectName}`;
+
+  // --yes accepts installing packages without prompting
+  const runCommand = runnerType === 'npx' ? `${runnerType} --yes` : runnerType;
+
+  if (process.env.NODE_ENV === 'development') {
+    toolbox.print.info(`${runCommand} nwui-cli@^0.4.1 add ${flags} ${finalComponents.join(' ')}`);
+  }
+
+  // @latest is getting cached when using bunx
+  await runSystemCommand({
+    command: `${runCommand} nwui-cli@^0.4.1 add ${flags} ${finalComponents.join(' ')}`,
```

Comment:
From my testing with `npx`, I'm pretty sure it won't check for a newer version if an previously installed version matches it... if so, I think we should change this PR to use the version specifier if `bun`/etc is selected, otherwise `latest`.

I could also see the argument that CES should not use `latest`, and instead let it change the version specifier separately once support is confirmed / working for it.

@danstepanov thoughts?

Reply:
Teah maybe specific versions is best since we can be sure specific combinations of ces versions and nwui versions work together

Reply:
Also yeah it might not look for newer versions but atleast the first time you have the latest matching version.

its mostly just to make sure there aren't any older versions being used